text/0036-ternary.md

```diff
@@ -0,0 +1,280 @@
+may change later, but broadly we intend to introduce an API that does not
+inherently inhibit high performance through poor design choices.
+
+# Detailed design
+
+`bee-ternary` is designed to be extensible and is built on top of a handful of
+fundamental abstraction types. Below we list the features of the API and
+summarize a few of the core API features.
+
+## Features
+
+- Efficient manipulation of ternary buffers (trits and trytes)
+- Multiple encoding schemes
+- Extensible design that allows it to sit on top of existing data structures,
+avoiding unnecessary allocation and copying
+- An array of utility functions to allow for easy manipulation of ternary data
+- Zero-cost conversion between trit and tryte formats
+
+## Key Types & Traits
+
+### `Btrit` and `Utrit`
+
+```rust
+enum Btrit { NegOne, Zero, PlusOne }
+enum Utrit { Zero, One, Two }
+```
+
+The crate supports both balanced and unbalanced trit representations, supported
+through two types, `Btrit` and `Utrit`. `Btrit` is the more common
+representation and so is the implicit default for most operations in the crate.
+
+### `Trits`
+
+```rust
+struct Trits<T: RawEncoding> { ... }
+```
+
+- Analogous to `str` or `[T]`
+- Unsized type, represents a buffer of trits of a specified encoding
+- Most commonly used from behind a reference (as with `&str` and `&[T]`),
+representing a 'trit slice'
+- Can be created from a variety of types such as`TritBuf` and `[i8]` with
+minimal overhead
+- Sub-slices can be created from trit slices at a per-trit level
+
+### `TritBuf`
+
+```rust
+struct TritBuf<T: RawEncodingBuf> { ... }
+```
+
+- Analogous to `String` or `Vec<T>`
+- Most common way to manipulate trits
+- Allows pushing, popping, and collecting trits
+- Implements `Deref<Target=Trits>` (in the same way that `Vec<T>` implements
+`Deref<[T]>`)
+
+### `RawEncoding`
+
+```rust
+trait RawEncoding { ... }
+```
+
+- Represents a raw trit buffer of some encoding
+- Common interface implemented by all trit encodings (`T1B1`, `T3B1`, `T5B1`,
+etc.)
+- Largely an implementation detail, not something you need to care about unless
+implementing your own encodings
+- Minimal implementation requirements: safety and most utility functionality is
+provided by `Trits` instead
+
+### `RawEncodingBuf`
+
+```rust
+trait RawEncodingBuf { ... }
+```
+
+- Buffer counterpart of `RawEncoding`, always associated with a specific
+`RawEncoding`
+- Distinct from `RawEncoding` to permit different buffer-like data structures
+in the future (linked lists, stack-allocated arrays, etc.)
+
+### `T1B1`/`T2B1`/`T3B1`/`T4B1`/`T5B1`
+
+```rust
+struct TXB1 { ... }
+```
+
+- Types that implement `RawEncoding`
+- Allow different encodings in `Trits` and `TritBuf` types.
+
+### `T1B1Buf`/`T2B1Buf`/`T3B1Buf`/`T4B1Buf`/`T5B1Buf`
+
+```rust
+struct TXB1Buf { ... }
+```
+
+- Types that implement `RawEncodingBuf`
+- Allow different encodings in `TritBuf` types.
+- Each type is associated with a `RawEncoding` type.
```

Comment:
Tiny nit again, sorry about my OCD triggering, you sometimes end list items with a dot or not

Reply:
Will fix this